autoarray/dataset/imaging/dataset.py

```diff
@@ -28,25 +28,72 @@ def __init__(
         noise_map: Optional[Array2D] = None,
         psf: Optional[Kernel2D] = None,
         noise_covariance_matrix: Optional[np.ndarray] = None,
-        settings: SettingsImaging = SettingsImaging(),
+        sub_size: int = 1,  # Temporary before refactor
+        sub_size_pixelization: int = 4,  # Temporary before refactor
+        over_sample: Optional[OverSampleIterate] = None,
+        over_sample_pixelization: Optional[OverSampleIterate] = None,

```

Comment:
I'm not sure it's necessarily useful here but it's worth knowing about using Flags as a way to passing settings.

```Python
from enum import Flag

class OverSample(Flag):
    iterate = 1
    iterate_pixelisation = 2

def do_thing(over_sample: OverSample):
    if OverSample.iterate in over_sample:
        ...
    if OverSample.iterate_pixelisation in over_sample:
        ...

# You can pass one value to a function
do_thing(OverSample.iterate)

# Or pass multiple
do_thing(OverSample.iterate | OverSample.iterate_pixelisation)